tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +214,97 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	type watcher struct {
+		prefix string
+		c      *clientv3.Client
+		wc     clientv3.WatchChan
+	}
+	prefixes := 36
+	watchPerKey := 10
+	ctx := context.Background()
+	prefixKeysMap := make(map[string][]string)
+	ws := make([]watcher, prefixes*watchPerKey)
+
+	var (
+		wg  sync.WaitGroup
+		pr  *clientv3.PutResponse
+		err error
+	)
+
+	const (
+		chars = "abcdefghijklmnopqrstuvwxyz0123456789"
+	)
+
+	wg.Add(len(ws))
+	for i := 0; i < len(ws); {
+		for _, prefix := range chars {
+			go func(w *watcher, i int, prefix rune) {
+				defer wg.Done()
+				w.c = randClient(eps)
+				w.prefix = string(prefix)
+				w.wc = w.c.Watch(ctx, w.prefix, clientv3.WithPrefix())
+			}(&ws[i], i, prefix)
+			i++
+		}
+	}
+	wg.Wait()
+
+	done := make(chan bool)
+	go func(strlen int, runningTime float64, reqRate int, done chan bool) {
+		client := randClient(eps)
+		defer client.Close()
+		rand.Seed(time.Now().UTC().UnixNano())
+		result := make([]byte, strlen)
+		start := time.Now()
+		reqStart := time.Now()
+		reqCount := 0
+		for time.Since(start).Seconds() < runningTime {
+			for i := 0; i < strlen; i++ {
+				result[i] = chars[rand.Intn(len(chars))]
+			}
+			key := string(result)
+			pr, err = client.Put(ctx, key, key)
+			if err != nil {
```

Comment:
we need to handle error. probably just return until we see no error. we will test runner against a failure injected cluster. so we do expect timeouts or message lost.